xm-commons-domain-event/src/main/java/com/icthh/xm/commons/domain/event/domain/Outbox.java

```diff
@@ -0,0 +1,115 @@
+package com.icthh.xm.commons.domain.event.domain;
+
+import com.icthh.xm.commons.domain.event.domain.converter.MapToStringConverter;
+import com.icthh.xm.commons.migration.db.jsonb.Jsonb;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.data.domain.Persistable;
+
+import javax.persistence.AttributeOverride;
+import javax.persistence.AttributeOverrides;
+import javax.persistence.Column;
+import javax.persistence.Convert;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+@Entity
+@Table(name = "outbox")
+@Getter
+@Setter
+public class Outbox implements Serializable, Persistable<UUID> {
+
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(generator = "UUID")
+    @Column(name = "id", updatable = false, nullable = false)
+    private UUID id;
+
+    @NotNull
+    @Column(name = "tx_id")
+    private String txId;
```

Comment:
@vkirichenko @valeriibilenko 
Regarding `txid` and its usage:
The main idea that we need to put here real database transaction ID. It will help to understand all entities mutated together in one tranzation.

As I understant this transaction Id is fully DB-specific and vcan be obtained only with native SQL.
Here you can find appropriate commands:
https://vladmihalcea.com/current-database-transaction-id/

I see next implementation notes here:
1. Ideally it should be configurable feature to use real TXID from the DB or let calling party to pit it's own.
2. We can implement TXId resolver similaryl as we did here: `com.icthh.xm.commons.migration.db.tenant.SchemaChangeResolver`
3. To avoid selecting current transation ID from DB multiple times (for every event generated within transaction) it proposed to add some threatlocal / atomic reference to hold it fetch only once with first domain event.